### DISCUSSION TITLE
Network hardware emulation

### DISCUSSION THREAD

**Discussion by emuslor** [86Box/86Box#3265] [Q&A] [answered]
Are there notable differences in network hardware emulation between PCem v17 and 86box 3.11?  I'm really digging 86box, and only one hurdle I can't seem to get across...  Something about the networking setup in my environment is giving apparent hangs at some level when the network receives more than a few KB in a short time.  I've tried to provide an example below using similar hardware in both emulators, trying a few different network cards (with my primary being NE2000 since that's what was working well in PCem), packet drivers, and mTCP as the TCP/IP stack.  If you check the two images below, you'll see I was able to get to the point of downloading a file from funet in both cases, but with 86box it does not complete the transfer when it seemingly gets to the full size, at which point can send a ctrl-alt-delete to reboot.  After reboot, the file on the virtual disk is zero bytes.  Also, just doing a listing of a large directory can cause the hang to happen.  Other things with generally smaller amounts of traffic, like say ircjr, seem to do ok.  Any thoughts what might be going on and how I might get around it?  Also attached is my 86box config.  Both tests used the same virtual disk.

![image](https://user-images.githubusercontent.com/1514398/233679792-4d98addc-382f-4739-b16e-cff976db23b7.png)

[86box.txt](https://github.com/86Box/86Box/files/11297315/86box.txt)

Thanks!

**Comment by Carmina16** [comment, marked as answer]
Looks like #3199 .

**Reply by emuslor** [reply]
Oh yeah, it sure does.  Thanks!
